akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -354,4 +389,51 @@ object HttpEntity {
     def getTrailerHeaders: JIterable[jm.HttpHeader] = trailer.asJava
   }
   object LastChunk extends LastChunk("", Nil)
+
+  /**
+   * Turns the given source into one that respects the `withSizeLimit` calls when used as a parameter
+   * to entity constructors.
+   */
+  def limitable[Out: SizeOfInBytes, Mat](source: Source[Out, Mat]): Source[Out, Mat] =
+    source.transform { () ⇒
+      new PushStage[Out, Out] {
+        val sizeOf = implicitly[SizeOfInBytes[Out]]
```

Comment:
Does the context-bound on `Out` suffice?
Or maybe context-bounds are against the Akka style guide? (I don't quite remember anything here, but maybe things have changed as well...)